include/gutter_tree.h

```diff
@@ -123,11 +121,15 @@ class BufferTree {
    */
   flush_ret_t force_flush();
 
+  // Functions for flushing bcbs or subtrees of the graph
+  flush_ret_t flush_subtree(flush_struct &flush_from, buffer_id_t first_child);
```

Comment:
why can't we include `first_child` in the `flush_struct`?

Reply:
unless I'm misunderstanding the purpose of `flush_struct`

Reply:
flush_struct is on a per flush_thread basis whereas first_child (now renamed to root_id) is the id of the root. There may be many more roots than threads.

Reply:
so flush_struct correspond to physical threads? (either way can u document?)

Reply:
Done